ethpm/validation.py

```diff
@@ -0,0 +1,64 @@
+import re
+
+from urllib import parse
+
+from eth_utils import is_checksum_address
+
+from ethpm.constants import (
+    PACKAGE_NAME_REGEX,
+    REGISTRY_URI_SCHEME,
+)
+
+from ethpm.exceptions import UriNotSupportedError
+
+from ethpm.utils.registry import (
+    is_ens_domain,
+)
+
+
+def validate_package_name(pkg_name: str) -> None:
+    """
+    Raise an exception if the value is not a valid package name
+    as defined in the EthPM-Spec.
+    """
+    if not bool(re.match(PACKAGE_NAME_REGEX, pkg_name)):
+        raise UriNotSupportedError('{0} is not a valid package name.'.format(pkg_name))
+
+
+def validate_registry_uri(uri: str) -> None:
+    """
```

Comment:
I love that you focus on landing the APIs with docs right from the start :heart:

Reply:
Danke!! learning from the best